src/component/ItemDetails.js

```diff
@@ -20,20 +40,41 @@ const ItemDetails = ({ token, purchased }) => {
         }
 
         // Below lines calculate next purchase date using dayjs 'add' function.
-        const lastPurchaseDate = item.dateOfPurchase.toDate();
-        const nextPurchaseDate = dayjs(lastPurchaseDate)
-          .add(item.numberOfDays.toString(), 'day')
-          .toDate();
+        let lastPurchaseDate;
+        let nextPurchaseDate;
+
+        if (item.dateOfPurchase) {
+          lastPurchaseDate = item.dateOfPurchase.toDate();
+          nextPurchaseDate = dayjs(lastPurchaseDate)
+            .add(item.numberOfDays.toString(), 'day')
+            .toDate();
+
+          setIsNewItem(false);
+        }
 
         return (
           <main>
             <BackButton />
             <h1>{item.name}</h1>
             <ul className="itemDetails">
-              <li>Last purchase: {lastPurchaseDate.toDateString()}</li>
-              <li>Next purchase: {nextPurchaseDate.toDateString()}</li>
+              <li>
+                Last purchase:{' '}
```

Comment:
This looks so noiiiice. I wish my brain was this clever.

Reply:
I'm right there with you @michellejae 😂

Reply:
Agreed this looks so clean and easy to follow! Nice job guys.

Reply:
🙌 awesome work!